cminx/cminx.cmake

```diff
@@ -0,0 +1,32 @@
+set(CMAKEDOC_VENV "${VENV}")
```

Comment:
You should be using the Python interpreter that `find_package(Python3 ...)` finds. For the purposes of the calls to these functions which happens in `CMinx/CMakeLists.txt` you know that you already called `find_package(Python3 ...)` and the interpreter is available. We can worry about ensuring that it's called by downstream dependencies later.

Reply:
As far as I'm aware we need to use the virtual environment python, not the one found by FindPython because the dependencies are not installed system-wide, only in the virtual environment. I can add an option to install system wide instead though

Reply:
If you want CMake to use the virtual environment Python interpreter I think you can just set `Python3_ROOT_DIR` to point to the virtual environment directory when you run cmake.

Reply:
Yeah I'd blow away your build directory before doing it just to make sure it doesn't used the cached value. Also to be clear I mean to use that variable like: `cmake -DPython3_ROOT_DIR=<root/dir/goes/here>`. Also you'll need to update your `execute_process` command so that it uses `Python3_EXECUTABLE`.

Reply:
Oh I see what you mean now. I used the virtual environment approach just for building CMinx, at install time it's packaged into a single file so it doesn't clutter up the system packages, if we use the root dir like that then by default it will be using the system python and clutter up the packages anyways. So how it goes is:

Configure:
1. Find system python, setuptools, and venv
2. Construct virtual environment in build dir by calling system python
3. Install dependencies in venv
4. Use venv to generate documentation. Can't use single binary because it's not compiled yet.

Build:
1. Optionally build docs using sphinx

Install:
1. Compile dependencies from venv and main python file into single executable
2. Copy executable to installation_prefix/bin

I can modify it to instead install packages using the root dir and not compile a single binary if you'd like

Reply:
I forgot all this happened under the hood. Disregard my comments and switch back to the virtual environment Python. Sorry.